src/main/scala/org/apache/spark/sql/hive/llap/LlapContext.scala

```diff
@@ -124,6 +122,44 @@ class LlapContext(sc: SparkContext,
       Seq.empty
     }
   }
+
+  override protected[sql] def executePlan(plan: LogicalPlan): super.QueryExecution = {
+    new this.QueryExecution(plan)
+    super.executePlan(plan)
+  }
+
+  private[sql] class QueryExecution(logicalPlan: LogicalPlan)
+    extends super.QueryExecution(logicalPlan)  {
+
+    /**
+     * Access control for Desc Table Function
+     */
+    val checkDesc = executedPlan match {
+      case ExecutedCommand(desc: DescribeCommand) =>
+        var database = ""
+        var tablename = ""
+        val args = desc.argString
+        args.split(",").foreach{
+          s =>
+            if (s.contains("table"))
+              {
+                val tables = s.split("->")
+                val nameParts = tables{1}.split("\\.")
+                if (nameParts.length != 2) {
```

Comment:
That is the correct and consistent behavior with Apache Spark and also with SPARK-LLAP 2.x. :)